sktime/forecasting/base/adapters/_lagllama.py

```diff
@@ -0,0 +1,626 @@
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from lag_llama.gluon.estimator import LagLlamaEstimator
+
+        # Creating a new LagLlama estimator with the appropriate
+        # forecasting horizon
+        self.estimator_ = LagLlamaEstimator(
+            ckpt_path="lag-llama.ckpt",
+            prediction_length=32 if not fh else fh,  # This is the most important here!
+            context_length=self.context_length_,
+            input_size=self.estimator_args["input_size"],
+            n_layer=self.estimator_args["n_layer"],
+            n_embd_per_head=self.estimator_args["n_embd_per_head"],
+            n_head=self.estimator_args["n_head"],
+            scaling=self.estimator_args["scaling"],
+            time_feat=self.estimator_args["time_feat"],
+            batch_size=self.batch_size_,
+            device=self.device_,
+            lr=self.lr_,
+            trainer_kwargs=self.trainer_kwargs_,
+        )
+
+        lightning_module = self.estimator.create_lightning_module()
+        transformation = self.estimator.create_transformation()
+
+        # Creating a new predictor
+        self.predictor_ = self.estimator_.create_predictor(
+            transformation, lightning_module
+        )
+
+        # Lastly, training the model
+        self.predictor_ = self.estimator.train(y.train, cache_data=True)
+
+    def _predict(self, fh, X):
+        """Forecast time series at future horizon.
+
+        private _predict containing the core logic, called from predict
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon is not read here, please set it as
+            `prediction_length` during initialization or when calling the fit function
+
+        X : GluonTS PandasDataset Object (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+
+        Returns
+        -------
+        y_pred : GluonTS PandasDataset Object
+            should be of the same type as seen in _fit, as in "y_inner_mtype" tag
+            Point predictions
+        """
+        from gluonts.evaluation import make_evaluation_predictions
+
+        # Creating a forecaster object
+        forecast_it, _ = make_evaluation_predictions(
+            dataset=X, predictor=self.predictor_, num_samples=self.num_samples_
+        )
+
+        # Forming a list of the forecasting iterations
+        forecasts = list(forecast_it)
+
+        return forecasts
+
+    # todo: consider implementing this, optional
+    # if not implementing, delete the _update method
+    def _update(self, y, X=None, update_params=True):
```

Comment:
I would propose to not implement update for know. Since it is not clear how it should look like for global forecasting and foundation model in sktime.